### PR TITLE
Replace identifier type inputs with select elements

### PR DESCRIPTION
**Pull request by daniellemaxwell**
### Scope of changes

- Displays a select element in the transaction accept form to allow users to edit an identifier type.
- Replaces country objects with imported countries array from `constants.js`.
- Adds empty value handling to the `TransactionDetail` partial.

### Type of change

- [ ] bug fix
- [ ] new feature
- [ ] documentation
- [x] other (describe)

### Acceptance criteria

Does each `select` element have the same class name?
Does each input type storing the identifier type value have `type="hidden"`?

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ] I have updated the dependencies list
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have added or updated the documentation
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] To the best of my ability, I believe that this PR represents a good solution to the specified problem and that it should be merged into the main code base.

### CODE REVIEW

File: pkg/web/static/js/countrySelect.js [hunk 1]
```diff
@@ -1,269 +1,22 @@
...
+// Display a searchable dropdown for countries in the originator section of the send envelope form.
 const origCountrySelect = new SlimSelect({
   select: '#orig_countries',
   settings: {
     contentLocation: document.getElementById('orig-countries-content'),
   }
 });
 
-// Can more than one country be selected?
+// Display a searchable dropdown for countries in the beneficiary section of the send envelope form.
```
Comment (daniellemaxwell) [inline pkg/web/static/js/countrySelect.js:11]: I decided to keep these separate for now, but could update the current implementation to use `querySelector`.

File: pkg/web/static/js/transactionAcceptPreview.js [hunk 2]
```diff
@@ -28,14 +28,51 @@ document.body.addEventListener('htmx:afterSettle', () => {
...
+    const selectedIdentifier = document.querySelector(`.${identifierID}`)
+    if (selectedIdentifier) {
+      setIdentifierData(identifier, selectedIdentifier)
+    }
+  })
 })
 
 // Set the country options and selected value in a SlimSelect dropdown for each country selection.
-function setSlimData(el, value) {
+function setCountryData(el, value) {
```
Comment (pdamodaran) [inline pkg/web/static/js/transactionAcceptPreview.js:52]: Good call on changing the function name to be more specific!

File: pkg/web/static/js/counterparties.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import { countriesArray } from './constants.js';
```
Comment (pdamodaran) [inline pkg/web/static/js/counterparties.js:1]: good call moving the countries list to a constants file to avoid duplication!

**Review by pdamodaran — APPROVED** [review]
Nice work @daniellemaxwell!

**Pull request merged**